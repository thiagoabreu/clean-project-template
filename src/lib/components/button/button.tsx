import React, { FC } from 'react';

import style from './style.module.css';

type SimpleButtonProps = {
  onClick: () => void;
};

export const SimpleButton: FC<SimpleButtonProps> = (props) => {
  return (
    <button className={style.activeButton} onClick={props.onClick}>
      {props.children}
    </button>
  );
};
