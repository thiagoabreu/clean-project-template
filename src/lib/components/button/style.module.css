.activeButton {
  border: 4px solid var(--primary-color);
}
