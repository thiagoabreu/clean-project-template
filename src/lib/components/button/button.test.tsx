import React from 'react';

import { SimpleButton } from './button';
import { render } from '@testing-library/react';

describe('<SimpleButton/>', function () {
  it('should render correctly', function () {
    const { container } = render(<SimpleButton onClick={() => null} />);

    expect(container).toMatchSnapshot();
  });
});
