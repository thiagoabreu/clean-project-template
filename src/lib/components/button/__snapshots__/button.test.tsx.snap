// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleButton/> should render correctly 1`] = `
<div>
  <button />
</div>
`;
